### ISSUE TITLE
Strange slide animation

### ISSUE THREAD

**Issue by Josephjbliss** [xiaolin/react-image-gallery#138]
It seems when you jump to or from slide 1 from a slide past slide 3, you can see the last slide animate in just behind slide 1 but in front of the rest of the slides. I think it's a `z-index` thing ...

**Comment by Josephjbliss** [comment]
Here's an example slowed way down: 

!(image)[http://i.giphy.com/l3q2ZwcPw20d5NNJK.gif]

**Comment by Josephjbliss** [comment]
Here is an example of the behavior: 

![image](http://i.giphy.com/l3q2ZwcPw20d5NNJK.gif)

**Comment by xiaolin** [comment]
@Josephjbliss I notice this when you set the `slideInterval` very high, I'll look into this when I get a chance.

**Comment by Josephjbliss** [comment]
Thank you! I will also take a look at a solution / PR. It's especially noticeable when it's super high. I currently have it set to only `800ms` and it is problematic, though.

**Comment by xiaolin** [comment]
FYI, as an immediate fix, if you are okay with using  `infinite={false}` it would make the problem go away.

**Comment by xiaolin** [comment]
@kamui @Josephjbliss I have a solution by adding a lower zIndex on the first and last slide, and adding a lock to prevent further `slideToIndex` based on `slideDuration`

```
    this._lockSliding = true;
    window.setTimeout(() => {
      this._lockSliding = false;
    }, this.props.slideDuration);
```

What this ultimately does is prevent the user from clicking to a new slide until the next slide finishes its animation.

Do you guys think this is okay?

**Comment by kamui** [comment]
Thanks! I'll take a look when I get a chance. Do you plan on releasing a new version soon?

**Comment by xiaolin** [comment]
Yes, I will release this version tonight.

**Comment by xiaolin** [comment]
@kamui @Josephjbliss released in 0.7.11
